packages/dev/core/src/Rendering/IBLShadows/iblShadowsSpatialBlurPass.ts

```diff
@@ -199,6 +203,10 @@ export class _IblShadowsSpatialBlurPass {
             width: Math.max(1.0, Math.floor(this._engine.getRenderWidth() * scaleFactor)),
             height: Math.max(1.0, Math.floor(this._engine.getRenderHeight() * scaleFactor)),
         };
+        // Don't resize if the size is the same as the current size.
+        if (this._outputTexture.getSize().width === newSize.width && this._outputTexture.getSize().height === newSize.height) {

```

Comment:
I also added logic to prevent resizing if the size hasn't actually changed. I assume that resizing the procedural texture may result in the shader program needing to be recreated so I was trying to prevent this from happening.

Reply:
No, resizing the procedural texture does not trigger a recompilation of the shader, but it is better not to resize the texture if it is not necessary.